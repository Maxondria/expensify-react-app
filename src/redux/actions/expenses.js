import actionTypes from "../constants";
import uuid from "uuid";

const addExpense = ({
  description = "",
  note = "",
  amount = 0,
  createdAt = 0
} = {}) => ({
  type: actionTypes.ADD_EXPENSE,
  expense: {
    id: uuid(),
    description,
    note,
    amount,
    createdAt
  }
});

const removeExpense = (id) => ({
  type: actionTypes.REMOVE_EXPENSE,
  id
});

const editExpense = (id, expense) => ({
  type: actionTypes.EDIT_EXPENSE,
  expense
});

export { addExpense, removeExpense, editExpense };
