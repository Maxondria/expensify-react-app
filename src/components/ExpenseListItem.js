import React from "react";
import { connect } from "react-redux";
import { removeExpense } from "../redux/actions/expenses";

const ExpenseListItem = ({ description, amount, createdAt, id, dispatch }) => {
  return (
    <div>
      <h3>{description}</h3>
      <p>
        {amount} - {createdAt}
        <button onClick={() => dispatch(removeExpense(id))}>Remove</button>
      </p>
    </div>
  );
};

export default connect()(ExpenseListItem);
