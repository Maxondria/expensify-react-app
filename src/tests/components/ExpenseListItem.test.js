import React from "react";
import { shallow } from "enzyme";
import ExpenseListItem from "../../components/ExpenseListItem";
import expenses from "../redux/fixtures/expenses.fixture";

it("should render one expense item correctly", function() {
  const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);

  expect(wrapper).toMatchSnapshot();
});
