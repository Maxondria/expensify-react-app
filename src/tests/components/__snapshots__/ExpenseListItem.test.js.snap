// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render one expense item correctly 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <h3>
      Cheese - Cottage Cheese
    </h3>
  </Link>
  <p>
    71.97
     - 
    0
  </p>
</div>
`;
